src/hooks/use-is-complete/use-is-complete.spec.ts

```diff
@@ -0,0 +1,158 @@
+import { act, Renderer, renderHook, RenderHookResult } from '@testing-library/react-hooks';
+import {
+	asyncScheduler,
+	AsyncSubject,
+	BehaviorSubject,
+	ConnectableObservable,
+	EMPTY,
+	GroupedObservable,
+	NEVER,
+	Observable,
+	of,
+	ReplaySubject,
+	scheduled,
+	Subject,
+	throwError,
+} from 'rxjs';
+import { useIsComplete } from './use-is-complete';
+
+type useIsCompleteParams = Parameters<typeof useIsComplete>;
+type useIsCompleteReturn = ReturnType<typeof useIsComplete>;
+
+function renderUseIsComplete(
+	initialParams: useIsCompleteParams
+): RenderHookResult<useIsCompleteParams, useIsCompleteReturn, Renderer<useIsCompleteParams>> {
+	return renderHook((params: useIsCompleteParams) => useIsComplete(...params), {
+		initialProps: initialParams,
+	});
+}
+
+describe('useObservableState', () => {
+	it.each`
+		observable
+		${new Observable()}
+		${new ConnectableObservable(new Observable(), () => new Subject())}
+		${new GroupedObservable('', new Subject())}
+		${new Subject()}
+		${new BehaviorSubject('')}
+		${new ReplaySubject(0)}
+		${new AsyncSubject()}
+	`('initially returns false when given $observable', ({ observable }) => {
+		const { result } = renderUseIsComplete([observable]);
+		expect(result.current).toBe(false);
+	});
+
+	it.each`
+		value
+		${'value2'}
+		${{}}
+		${Number.MIN_VALUE}
+		${[]}
+		${undefined}
+		${null}
+		${new Date(2000, 4, 15, 9, 55)}
+	`(
+		'returns false when given a subject which emits $value after the initial render',
+		({ value }) => {
+			const subject = new Subject();
+			const { result } = renderUseIsComplete([subject]);
+			act(() => subject.next(value));
+			expect(result.current).toEqual(false);
+		}
+	);
+
+	it.each`
+		observable                     | expected
+		${of(1, 2, 3)}                 | ${[false, true]}
+		${of(false)}                   | ${[false, true]}
+		${of('a', 'b', 'c', 'd')}      | ${[false, true]}
+		${EMPTY}                       | ${[false, true]}
+		${NEVER}                       | ${[false]}
+		${throwError(new TypeError())} | ${[false, new TypeError()]}
+		${throwError('error')}         | ${[false, 'error']}
+	`('returns $expected when given $initialValue and $observable ', ({ observable, expected }) => {
+		const { result } = renderUseIsComplete([observable]);
+		expect(result.all).toEqual(expected);
+	});
+
+	it.each`
+		observable                     | expected
+		${of(1, 2, 3)}                 | ${[false, true, true, true]}
+		${of(false)}                   | ${[false, true, true, true]}
+		${of('a', 'b', 'c', 'd')}      | ${[false, true, true, true]}
+		${EMPTY}                       | ${[false, true, true, true]}
+		${NEVER}                       | ${[false, true, true, false]}
+		${throwError(new TypeError())} | ${[false, true, true, new TypeError()]}
+		${throwError('error')}         | ${[false, true, true, 'error']}
+	`(
+		'returns $expected when given an observable of [1, 10], then re-rendered with $observable',
```

Comment:
This test could be made a bit more obvious. The first call to `renderUseIsComplete` is supposed to represent an observable which emits immediately